client/txn.go

```diff
@@ -371,6 +387,9 @@ func (txn *Txn) CommitNoCleanup() *roachpb.Error {
 // optional, but more efficient than relying on the implicit commit
 // performed when the transaction function returns without error.
 // The batch must be created by this transaction.
+// If the command completes successfully, the txn is considered finalized. On
+// error, no attempt is made to clean up the (possibly still pending)
+// transaction.
```

Comment:
It's pretty inconsistent that `CommitInBatch` behaves differently from `Commit` with respect to cleanup. I think `CommitNoCleanup` ought to be renamed to `Commit` and that the current `Commit` should be `CommitOrRollback` (or your favorite name for attempt commit, attempt rollback if the commit fails).